### PR TITLE
[KeyVault] Breaking changes: soft delete and more

### LINKED ISSUE #10554: [KeyVault] Enable soft-delete by default when creation
## Description of the new feature
Soft-delete option will be enabled by default when creation

## Proposed implementation details (optional)
This would involve syntax breaking change:
Old: `[-EnableSoftDelete]`
New: `[-DisableSoftDelete]`

### LINKED ISSUE #10555: [KeyVault] Support for soft-delete on/off/null as a one time operation
## Description of the new feature

Keyvault team are changing the default behavior that when creating a new keyvault, it's soft-delete is on.
We need to make sure:

- [ ] The default behavior remains the same (soft delete is off) so as to keep it compatible
- [ ] Provide a one-time operation to enable soft delete (this cannot be undone)

## Proposed implementation details (optional)

Since the requirement is changing a property of the keyvault resource, idealy we should add an optional parameter `[-EnableSoftDelete]` to the `Update-AzKeyVault` cmdlet, but that cmdlet does not exist (only get- new- remove-), so we might need to add a `update-` cmdlet.

- .Net SDK
  - [ ] Update api version
  - [ ] Write scenario tests
- PowerShell
  - [ ] Utilize new SDK
  - [ ] Keep the current behavior of `New-AzKeyVault`
  - [ ] A new cmdlet: `Update-AzKeyVault`
  - [ ] Tests

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

This PR contains two breaking changes:
1. `[-EnableSoftDelete]` is replaced by `[-DiableSoftDelete]` (Fixes #10554). So the default behavior of `New-AzKeyVault` is now:
    - `New-AzKeyVault` -> soft delete enabled
    - `New-AzKeyVault -DisableSoftDelete` -> opt-out soft delete
1. Two cmdlet aliases are removed:
    - `New-AzKeyVaultCertificateAdministratorDetails`
    - `New-AzKeyVaultCertificateOrganizationDetails`

and one non-breaking change:
1. `New-AzKeyVault` and `Update-AzKeyVault` now supports `[-SoftDeleteRetentionInDays <int>]` (resolves #10555)

## Checklist
- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/542
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
The failures in the "analyze" jobs are expected as I AM making breaking changes.
Will suppress them later.

File: src/KeyVault/KeyVault/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Removed two aliases: `New-AzKeyVaultCertificateAdministratorDetails` and `New-AzKeyVaultCertificateOrganizationDetails`
```
Comment (isra-fel) [inline src/KeyVault/KeyVault/ChangeLog.md:21]: Caution: the first two line of `Upcoming Release` target S169, the rest target S168

**Comment by isra-fel** [comment]
The first commit was to format a 800-line ps1 file. Please exclude it when reviewing, or use this link: https://github.com/Azure/azure-powershell/pull/11469/files/4b298d915e341b1fdbd7fcd8b3156dbd4f50c781..5df1edf1fd47c3396b2e143c646d92e0e4ce7e33 . Thanks

**Pull request merged**